hybrasyl/Server.cs

```diff
@@ -67,12 +70,28 @@ public Server(int port)
             Port = port;
             PacketHandlers = new WorldPacketHandler[256];
             ControlMessageHandlers = new ControlMessageHandler[64];
+            Throttles = new Dictionary<byte, IThrottle>();
             ExpectedConnections = new ConcurrentDictionary<uint, Redirect>();
             for (int i = 0; i < 256; ++i)
                 PacketHandlers[i] = (c, p) => Logger.WarnFormat("Server: Unhandled opcode 0x{0:X2}", p.Opcode);
 
         }
 
+        public void RegisterThrottle(IThrottle newThrottle)
+        {
+            Throttles[newThrottle.Opcode] = newThrottle;

```

Comment:
I know this first pass is for packets only, but using throttle.opcode in the future might be a bad idea

Reply:
Maybe we should rename to IPacketThrottle and create a base interface for throttling which will be called IThrottle?

Reply:
I think that's a good idea, abstract from the base to create additional interfaces.